lib/src/map_extensions.dart

```diff
@@ -0,0 +1,8 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+extension MapExtensions<K, V> on Map<K, V> {
+  /// Like [Map.entries], but returns each entry as a record.
+  Iterable<(K, V)> get pairs => entries.map((e) => (e.key, e.value));
```

Comment:
Interesting...but why? What's the upside here? What do users get?

We're already going to allocate (however short-lived) `MapEnty` instances.

Reply:
The main benefit is being able to loop over pairs with variables named something other than "key" and "value".

https://github.com/dart-lang/collection/pull/289#issuecomment-1559974806

Reply:
I think we're still hoping to make `MapEntry` an inline class over a `(K, V)`, and if we are able to do that this `.pairs` extension will be useless. I think it's worth having it early in this package.

Reply:
I doubt it will be over `(K, V)`. More likely it's `({K key, V value})`, since that will allow `dynamic`-typed code to keep working.